### PR TITLE
Add Microsoft.RecoveryServices API schema

### PR DESCRIPTION
**Pull request by nilaydshah**
Microsoft.RecoveryServices API schema and tests for vaults resource type.

### CODE REVIEW

File: schemas/2016-06-01/Microsoft.RecoveryServices.json [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+              ]
+            },
+            {
+              "$ref": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression"
+            }
+          ],
+          "description": "Gets or sets the sku name. Required for vault creation, optional for update. Possible values include: 'RS0'"
+        }
+      },
+      "tier": {
```
Comment (wenwu449) [inline schemas/2016-06-01/Microsoft.RecoveryServices.json]: is this at the correct level? is should be under "properties", right?

Comment (nilaydshah) [inline schemas/2016-06-01/Microsoft.RecoveryServices.json]: This is at correct level for our RP. We take sku at top level (outside the properties bag) and both name and tier are properties of sku.
If you see Microsoft.Storage is also following same - sku outside properties in 2016-01-01/Microsoft.Storage.json

Comment (wenwu449) [inline schemas/2016-06-01/Microsoft.RecoveryServices.json]: if "tier" is a property of "sku", it should be under "properties", not the same level, the "properties" in ln82 is a schema object property, not name of property.

Comment (nilaydshah) [inline schemas/2016-06-01/Microsoft.RecoveryServices.json]: @wenwu449 thanks for pointing out. Updated PR with the fix.

**Pull request merged**